modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts

```diff
@@ -304,9 +307,95 @@ export class LinkModalDialog
         const checkbox = Checkbox.create().setLabelText(i18n('dialog.link.formitem.openinnewtab')).setInputAlignment(
             InputAlignment.LEFT).build();
 
+        checkbox.hide();
+
+        checkbox.onRendered(() => {
+            const contentSelector = (<ContentComboBox<ContentTreeSelectorItem>>this.getFieldById(
+                'contentId'));
+            const contentSelectorSelectedContent = contentSelector.getSelectedContent();
+
+            const radio = <RadioGroup>this.getFieldById('contentMediaRadio');
+            const radioValue = radio.doGetValue();
+
+            contentSelector.onValueChanged(() => {
+                const contentSelectorSelectedContent = contentSelector.getSelectedContent();
+
+                if (contentSelectorSelectedContent) {
+                    checkbox.show();
+                } else {
+                    checkbox.hide();
+                }
+            });
+
+            radio.onValueChanged((event: ValueChangedEvent) => {
+                if (event.getNewValue() === MediaContentRadioAction.LINK) {
+                    checkbox.show();
+                } else {
+                    checkbox.hide();
+                }
+            });
+
+            if (!radio.isVisible() && contentSelectorSelectedContent) {
+                checkbox.show();
+                return;
+            }
+
+            if (radioValue === MediaContentRadioAction.LINK) {
+                checkbox.show();
+                return;
+            }
+
+        });
+
         checkbox.setChecked(this.getTarget(isTabSelectedFn.call(this)));
 
         const formItemBuilder = new ModalDialogFormItemBuilder(id).setInputEl(checkbox);
+
+        return this.createFormItem(formItemBuilder);
+    }
+
+    private createRadio(id: string): FormItem {
+        const contentSelector = (<ContentComboBox<ContentTreeSelectorItem>>this.getFieldById(
+            'contentId'));
+
+        let formRadio: JQuery<HTMLElement>;
+        const radio = new RadioGroup('radio');
+
+        radio.addOption(MediaContentRadioAction.OPEN, i18n('dialog.link.radio.options.open'));
+        radio.addOption(MediaContentRadioAction.DOWNLOAD, i18n('dialog.link.radio.options.download'));
+        radio.addOption(MediaContentRadioAction.LINK, i18n('dialog.link.radio.options.link'));
+
+        radio.onRendered(() => {
+            formRadio =  $(radio.getHTMLElement()).closest('.input-view');
+            if (this.isInlineLink()) {
+                radio.setValue(MediaContentRadioAction.OPEN);
+            } else if (this.isDownloadLink()) {
+                radio.setValue(MediaContentRadioAction.DOWNLOAD);
+            } else if (this.isContentLink()) {
+                radio.setValue(MediaContentRadioAction.LINK);
+            } else {
+                radio.setValue(MediaContentRadioAction.OPEN);
+                formRadio.hide();
+            }
+        });
+
+        contentSelector.onValueChanged(() => {
```

Comment:
This should also be moved to `createSelector`. If the reason for assigning an event listener to, for example, a content selector inside `createRadio` method is to be able to easily get a hold of the radio control, then don't hesitate to assign controls to class-wide properties.

Schematic example:

```Typescript
private mediaOptionsRadioGroup: FormItem;
private contentSelector: FormItem;

createMediaOptionsRadioGroup() {
  this.mediaOptionsRadioGroup = ...;
  this.mediaOptionsRadioGroup.whenRendered = () => {
      this.contentSelector.doBlaBla...
  }
}

createContentSelector() {
  this.contentSelector = ...;
  this.contentSelector.whenRendered = () => {
      this.mediaOptionsRadioGroup.doBlaBla...
  }
}
```